src/WorkflowStub.php

```diff
@@ -110,4 +126,33 @@ private function dispatch()
 
         $this->model->class::dispatch($this->model, ...unserialize($this->model->arguments));
     }
+
+    public function __call($method, $arguments)
+    {
+        if (collect((new ReflectionClass($this->model->class))->getMethods())
+            ->filter(function ($method) {
+                return collect($method->getAttributes())
+                    ->contains(function ($attribute) {
+                        return $attribute->getName() === SignalMethod::class;
+                    });
+            })
+            ->map(function ($method) {
+                return $method->getName();
+            })->contains($method)
+        ) {
+            $this->model->signals()->create([
+                'method' => $method,
+                'arguments' => serialize($arguments),
+            ]);
+
+            while ($this->running()) {
+                try {
+                    $this->fresh()->dispatch();
+                    break;
+                } catch (\Spatie\ModelStates\Exceptions\TransitionNotFound $th) {
+                    usleep(1000);
```

Comment:
I'm going to get rid of this sleep/loop with a job that will retry by releasing itself back to the queue.